solana/rpc/providers/ws.py

```diff
@@ -0,0 +1,147 @@
+            self.ws = None
+
+
+class WSProvider(BaseProvider, FriendlyJsonSerde):
+    """WSProvider class for websockets."""
+
+    logger = logging.getLogger("solana.providers.WebsocketProvider")
+    _loop = None
+
+    def __init__(
+        self,
+        endpoint_uri: Union[URI, str],
+        websocket_kwargs: Optional[Any] = None,
+        websocket_timeout: int = DEFAULT_WEBSOCKET_TIMEOUT,
+    ) -> None:
+        """Initialize the websocket provider."""
+        self._request_counter = itertools.count()
+        if isinstance(endpoint_uri, str):
+            self.endpoint_uri = URI(endpoint_uri)
+        else:
+            self.endpoint_uri = endpoint_uri
+        self.websocket_timeout = websocket_timeout
+        if self.endpoint_uri is None:
+            self.endpoint_uri = get_default_endpoint()
+        if WSProvider._loop is None:
+            WSProvider._loop = _get_threaded_loop()
+        if websocket_kwargs is None:
+            websocket_kwargs = {}
+        else:
+            found_restricted_keys = set(websocket_kwargs.keys()).intersection(RESTRICTED_WEBSOCKET_KWARGS)
+            if found_restricted_keys:
+                raise Exception(
+                    "{0} are not allowed in websocket_kwargs, "
+                    "found: {1}".format(RESTRICTED_WEBSOCKET_KWARGS, found_restricted_keys)
+                )
+        self.conn = PersistentWebSocket(self.endpoint_uri, WSProvider._loop, websocket_kwargs)
+
+    def __str__(self) -> str:
+        """Return a string representation of the websocket connection."""
+        return "WS connection {0}".format(self.endpoint_uri)
+
+    async def coro_make_request(self, request_data: bytes) -> RPCResponse:
+        """Makes coroutine requests."""
+        try:
+            async with self.conn as conn:
+                await asyncio.wait_for(conn.send(request_data), timeout=self.websocket_timeout)
+                return json.loads(await asyncio.wait_for(conn.recv(), timeout=self.websocket_timeout))
+        except TimeoutError:
+            self.logger.debug("Websocket connection timed out.")
+            raise
+        except Exception as err:
+            self.logger.debug("Unexpected Exception: %s", err)
+            raise
+
+    def make_request(self, method: RPCMethod, *params: Any) -> RPCResponse:
+        """Make a request to the rpc endpoint."""
+        request_id = next(self._request_counter) + 1
+        self.logger.debug("Making request WebSocket URI: %s, Method: %s, id: %s", self.endpoint_uri, method, request_id)
+        request_data = self.json_encode({"jsonrpc": "2.0", "id": request_id, "method": method, "params": params})
+        if WSProvider._loop:
+            future = asyncio.run_coroutine_threadsafe(
+                self.coro_make_request(cast(bytes, request_data)), WSProvider._loop
+            )
+        return future.result()
+
+    def connection(self) -> websockets.WebSocketClientProtocol:
```

Comment:
```suggestion
    def connect(self) -> websockets.WebSocketClientProtocol:
```

To be consistent with https://websockets.readthedocs.io/en/stable/api.html#module-websockets.client